src/Caller.cpp

```diff
@@ -707,17 +645,270 @@ bool Caller::parseOptions(int argc, char **argv) {
       cerr << "\nInvoke with -h option for help.\n";
       return 0; // ...error
     }
+
+    decoy_prefix = detect_decoy_prefix(inputFN_);
+    
   }
   // if there is more then one argument left...
   if (cmd.arguments.size() > 1) {
-    cerr << "Error: too many arguments.";
-    cerr << "\nInvoke with -h option for help\n";
-    return 0; // ...error
+    /* Check decoy-prefix */
```

Comment:
This is just too much code for a general routine. Could you please split out the decoy prefex checks etc? You preferably want it to sit in a section of the code dealing with decoys, e.g. SetHandler.